### PR TITLE
Expand docstrings, fix typos, fix paths.

### PR DESCRIPTION
**Pull request by cmarqu**
Refs #811.

### CODE REVIEW

File: tests/test_cases/test_cocotb/test_cocotb.py [hunk 1]
```diff
@@ -645,11 +645,11 @@ def test_clock_cycles(dut):
 
     yield RisingEdge(clk)
 
-    dut.log.info("After one edge")
+    dut._log.info("After one edge")
 
     yield ClockCycles(clk, 10)
 
-    dut.log.info("After 10 edges")
+    dut._log.info("After 10 edges")
```
Comment (eric-wieser) [inline tests/test_cases/test_cocotb/test_cocotb.py]: How did this work before?

File: cocotb/triggers.py [hunk 2]
```diff
@@ -459,7 +459,7 @@ def retval(self):
         return self._coroutine.retval
 
     def prime(self, callback):
-        """FIXME: document"""
+        """Don't call this in your own code, this is for internal use by the scheduler."""
```
Comment (eric-wieser) [inline cocotb/triggers.py]: I have a local branch with better docstrings than this, but haven't pushed it as an attept not to swamp @imphil with PRs

File: tests/test_cases/test_cocotb/test_cocotb.py (hunk 1 above)
Comment (cmarqu) [reply tests/test_cases/test_cocotb/test_cocotb.py]: Due to this: https://github.com/potentialventures/cocotb/blob/ed3e6d2dd4e870c40462d4385cc27779979bc961/cocotb/handle.py#L66

File: cocotb/triggers.py (hunk 2 above)
Comment (eric-wieser) [reply cocotb/triggers.py]: Edit: gh-1019

File: documentation/source/quickstart.rst [hunk 3]
```diff
@@ -87,22 +87,22 @@ test.
 
 .. code-block:: bash
 
-    $> export PATH=/opt/pym32/bin
-    $> cd <cocotb_dir>
+    $> export PATH=/opt/pym32/bin:$PATH
+    $> cd `cocotb-config --prefix`
     $> ARCH=i686 make
```
Comment (imphil) [inline documentation/source/quickstart.rst]: Two comments here:
- That whole 32 bit Linux section is a bit out of place. It's probably only needed for the free version of ModelSim? Can we move it to a more modelsim-specific page? I don't really want to start the "quickstart" documentation by a obscure edge case/workaround.
- The example how to use cocotb is out of place (cocotb is only installed further down in the documentation), and not really accurate either.

Comment (cmarqu) [reply documentation/source/quickstart.rst]: Maybe we could move that 32-bit section to https://github.com/cocotb/cocotb/wiki/Tier-2-Setup-Instructions and just add a link in the docs.

I'll try to clean up the rest a bit while trying to resist the urge to restructure anything.
A restructuring should be done in a different issue/PR, I'd like to follow https://www.divio.com/blog/documentation/ for that.

Comment (imphil) [reply documentation/source/quickstart.rst]: Moving to tier 2: sounds like a good plan.

I agree with you, all other things can come later.

**Comment by imphil** [comment]
Please rebase this on top of master to incorporate the changes done by Eric.

File: documentation/source/quickstart.rst (hunk 3 above)
Comment (cmarqu) [reply documentation/source/quickstart.rst]: I have re-arranged Quickstart a bit, please review.

Someone with Windows experience should improve this section - is e.g. Msys 1.0 still correct? And I guess using Anaconda/Miniconda would be the way to go these days? @themperek

Comment (cmarqu) [reply documentation/source/quickstart.rst]: The diff is hard to read, here is a rendered version: https://cocotb-cmarqu.readthedocs.io/en/fix-fixme-docstrings/quickstart.html

**Comment by eric-wieser** [comment]
@imphil: Good to go?

**Comment by imphil** [comment]
Manually rebased and merged.

**Pull request closed without merging**